openpype/hosts/houdini/api/lib.py

```diff
@@ -747,3 +753,64 @@ def get_camera_from_container(container):
 
     assert len(cameras) == 1, "Camera instance must have only one camera"
     return cameras[0]
+
+
+def update_houdini_vars_context():
+    """Update Houdini vars to match current context.
+
+    This will only do something if the setting is enabled in project settings.
+    """
+
+    project_settings = get_current_project_settings()
+    houdini_vars_settings = \
+        project_settings["houdini"]["general"]["update_houdini_var_context"]
+
+    if houdini_vars_settings["enabled"]:
+        houdini_vars = houdini_vars_settings["houdini_vars"]
+
+        # No vars specified - nothing to do
+        if not houdini_vars:
+            return
+
+        # Get Template data
+        template_data = get_current_context_template_data()
+
+        # Set Houdini Vars
+        for item in houdini_vars:
+
+            # For consistency reasons we always force all vars to be uppercase
+            item["var"] = item["var"].upper()
+
+            # get and resolve job path template
+            item_value = StringTemplate.format_template(
+                item["value"],
+                template_data
+            )
+
+            if item["is_path"]:
+                item_value = item_value.replace("\\", "/")
+                try:
+                    os.makedirs(item_value)
+                except OSError as e:
+                    if e.errno != errno.EEXIST:
+                        print(
+                            "  - Failed to create ${} dir. Maybe due to "
+                            "insufficient permissions.".format(item["var"])
+                        )
+
+            if item["var"] == "JOB" and item_value == "":
+                # sync $JOB to $HIP if $JOB is empty
+                item_value = os.environ["HIP"]
+
+            current_value = hou.hscript("echo -n `${}`".format(item["var"]))[0]
+
+            # sync both environment variables.
+            # because houdini doesn't do that by default
+            # on opening new files
+            os.environ[item["var"]] = current_value
+
+            if current_value != item_value:
+                hou.hscript("set {}={}".format(item["var"], item_value))
+                os.environ[item["var"]] = item_value
+
+                print("  - Updated ${} to {}".format(item["var"], item_value))
```

Comment:
```suggestion
    if not houdini_vars_settings["enabled"]:
        return
    
    houdini_vars = houdini_vars_settings["houdini_vars"]

    # No vars specified - nothing to do
    if not houdini_vars:
        return

    # Get Template data
    template_data = get_current_context_template_data()

    # Set Houdini Vars
    for item in houdini_vars:

        # For consistency reasons we always force all vars to be uppercase
        item["var"] = item["var"].upper()

        # get and resolve job path template
        item_value = StringTemplate.format_template(
            item["value"],
            template_data
        )

        if item["is_path"]:
            item_value = item_value.replace("\\", "/")
            try:
                os.makedirs(item_value)
            except OSError as e:
                if e.errno != errno.EEXIST:
                    print(
                        "  - Failed to create ${} dir. Maybe due to "
                        "insufficient permissions.".format(item["var"])
                    )

        if item["var"] == "JOB" and item_value == "":
            # sync $JOB to $HIP if $JOB is empty
            item_value = os.environ["HIP"]

        current_value = hou.hscript("echo -n `${}`".format(item["var"]))[0]

        # sync both environment variables.
        # because houdini doesn't do that by default
        # on opening new files
        os.environ[item["var"]] = current_value

        if current_value != item_value:
            hou.hscript("set {}={}".format(item["var"], item_value))
            os.environ[item["var"]] = item_value

            print("  - Updated ${} to {}".format(item["var"], item_value))
```

Reply:
Seems like this is now outdated - anyway the change would be:

Avoid the nested if branch
```python
if not enabled:
    return
```

Change comment:
```
# get and resolve job path template
```
to:
```
# get and resolve template in value
```